src/main/java/fi/aalto/cs/apluscourses/intellij/psi/PsiUtil.java

```diff
@@ -0,0 +1,20 @@
+package fi.aalto.cs.apluscourses.intellij.psi;
+
+import com.intellij.psi.PsiElement;
+import com.intellij.psi.PsiErrorElement;
+import com.intellij.psi.SyntaxTraverser;
+import org.jetbrains.annotations.NotNull;
+
+public class PsiUtil {
+  private PsiUtil() {
+
+  }
+
+  public static boolean psiSiblingHasErrors(@NotNull PsiElement element) {
+    return psiElementHasErrors(element.getPrevSibling()) || psiElementHasErrors(element.getNextSibling());
+  }
+
+  public static boolean psiElementHasErrors(@NotNull PsiElement element) {
+    return !SyntaxTraverser.psiTraverser(element).traverse().filter(PsiErrorElement.class).isEmpty();
+  }
+}
```

Comment:
Is this whole class ever used anywhere in the code

Reply:
it's for @stellatsv  to use later

Reply:
acknowledged